lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -501,9 +501,12 @@ def process_batch(
         else:
             extracted_pages = []
             with LogTime("text_extraction"):
-                for page in extractor_list:
-                    extracted_pages.append(text_extractor.extract_page(page))
-
+                for page_value in extractor_list:
+                    if isinstance(page_value, list):
+                        page = text_extractor.extract_page(page_data=page_value)
+                    else:
+                        page = text_extractor.extract_page(page_value)
+                    extracted_pages.append(page)
```

Comment:
This smells funny.  On the one hand, these two calls involve subclasses of the same class and have the same method name.  On the other hand, the calling conventions and inputs are totally different.  At the very least, the `page_data` thing looks like a hack.  I think the cleaner thing to do here is to have different method names when things should be called differently.

Reply:
Changed.